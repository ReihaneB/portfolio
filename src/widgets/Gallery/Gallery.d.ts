import type { ImageProps } from '@/components/Image/Image.d';

export interface GalleryProps {
  id?: number;
  name: string;
  images: ImageProps[];
}
