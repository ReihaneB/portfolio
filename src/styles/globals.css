:root {
  /* colors */
  --white: #f5f5f7;
  --cyan: #06b6d4;
  --blue: #2756df;
  --blue-lighter: #cee5ff;
  --blue-light: #dbeafe;
  --blue-dark: #1e40af;
  --focus-blue: #93c5fd;
  --grey: #6b7280;
  --grey-light: #e5e7eb;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  border: 0;
  font: inherit;
  vertical-align: baseline;
}

html,
body {
  max-width: 100vw;
  overflow-x: hidden;
}

details,
figcaption,
figure,
footer,
hgroup,
menu,
nav {
  display: block,
}

ol,
ul {
  list-style: none;
}

blockquote,
q {
  quotes: none;
}

blockquote::before,
blockquote::after,
q::before,
q::after {
  content: '';
  content: none;
}

table {
  border-collapse: collapse;
  border-spacing: 0;
}

a {
  color: inherit;
  text-decoration: none;
}

h1 {
  font-size: 30px;
  line-height: 38px;
  font-weight: bold;
}

h2 {
  font-size: 20px;
  line-height: 24px;
  font-weight: bold;
}

h3 {
  font-size: 16px;
  line-height: 20px;
  font-weight: bold;
}

h4 {
  font-size: 14px;
  line-height: 18px;
}

h5 {
  font-size: 12px;
  line-height: 16px;
}

p {
  font-size: 16px;
  line-height: 24px;
}