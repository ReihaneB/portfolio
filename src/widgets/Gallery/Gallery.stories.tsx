import type { Meta, StoryObj } from '@storybook/react';

import Gallery from './Gallery';

import styles from './Gallery.stories.module.css';

const meta: Meta<typeof Gallery> = {
  title: 'Widgets/Gallery',
  component: Gallery,
  decorators: [
    Story => (
      <div className={styles.root}>
        <Story />
      </div>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof Image>;

const defaultArgs = {
  name: 'Gallery',
  images: [
    {
      id: 1,
      src: 'https://via.placeholder.com/1920x1080',
      alt: 'Placeholder image',
    },
  ],
};

export const Playground: Story = {
  args: {
    ...defaultArgs,
  },
};

export const SquareImage: Story = {
  args: {
    ...defaultArgs,
    images: [
      {
        id: 1,
        src: 'https://via.placeholder.com/1080x1080',
        alt: 'Placeholder image',
      },
    ],
  },
};

export const _4x3Image: Story = {
  args: {
    ...defaultArgs,
    images: [
      {
        id: 1,
        src: 'https://via.placeholder.com/800x600',
        alt: 'Placeholder image',
      },
    ],
  },
};

export const _16x9Image: Story = {
  args: {
    ...defaultArgs,
    images: [
      {
        id: 1,
        src: 'https://via.placeholder.com/1280x720',
        alt: 'Placeholder image',
      },
    ],
  },
};

export const SingleImage: Story = {
  args: {
    ...defaultArgs,
    images: [
      {
        id: 1,
        src: 'https://via.placeholder.com/1920x1080',
        alt: 'Placeholder image',
      },
    ],
  },
};

export const MultipleImages: Story = {
  args: {
    ...defaultArgs,
    images: [
      {
        id: 1,
        src: 'https://via.placeholder.com/1920x1080',
        alt: 'Placeholder image',
      },
      {
        id: 2,
        src: 'https://via.placeholder.com/1920x1080',
        alt: 'Placeholder image',
      },
      {
        id: 3,
        src: 'https://via.placeholder.com/1920x1080',
        alt: 'Placeholder image',
      },
    ],
  },
};

export const LongName: Story = {
  args: {
    ...defaultArgs,
    name: 'Gallery with a really long name that should not be truncated',
  },
};

export const MultipleFormats: Story = {
  name: 'Multiple formats (1:1, 4:3, 16:9)',
  args: {
    ...defaultArgs,
    images: [
      {
        id: 1,
        src: 'https://via.placeholder.com/1080x1080',
        alt: 'Placeholder image',
      },
      {
        id: 2,
        src: 'https://via.placeholder.com/800x600',
        alt: 'Placeholder image',
      },
      {
        id: 3,
        src: 'https://via.placeholder.com/1280x720',
        alt: 'Placeholder image',
      },
    ],
  },
};
