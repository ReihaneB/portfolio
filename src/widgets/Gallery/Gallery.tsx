import Image from '@/components/Image/Image';

import type { GalleryProps } from './Gallery.d';
import styles from './Gallery.module.css';

function Gallery({
  name,
  images,
}: GalleryProps) {
  return (
    <>
      <h2 className={styles.name}>
        {name}
      </h2>
      <ul className={styles.images}>
        {images.map(({ id, src, alt }) => (
          <li key={id}>
            <Image
              src={src}
              alt={alt}
              priority
            />
          </li>
        ))}
      </ul>
    </>
  );
}

export default Gallery;
