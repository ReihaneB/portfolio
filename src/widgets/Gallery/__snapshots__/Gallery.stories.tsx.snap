// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widgets/Gallery _4x3Image smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-widgets-Gallery-Gallery-stories-module__root--O0_jV">
    <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
      Gallery
    </h2>
    <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
      <li>
        <img alt="Placeholder image"
             fetchpriority="high"
             width="0"
             height="0"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="https://via.placeholder.com/800x600?w=16&amp;q=75 1x"
             src="https://via.placeholder.com/800x600?w=16&amp;q=75"
             style="color: transparent;"
        >
      </li>
    </ul>
  </div>
</main>
`;

exports[`Widgets/Gallery _16x9Image smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-widgets-Gallery-Gallery-stories-module__root--O0_jV">
    <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
      Gallery
    </h2>
    <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
      <li>
        <img alt="Placeholder image"
             fetchpriority="high"
             width="0"
             height="0"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="https://via.placeholder.com/1280x720?w=16&amp;q=75 1x"
             src="https://via.placeholder.com/1280x720?w=16&amp;q=75"
             style="color: transparent;"
        >
      </li>
    </ul>
  </div>
</main>
`;

exports[`Widgets/Gallery LongName smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-widgets-Gallery-Gallery-stories-module__root--O0_jV">
    <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
      Gallery with a really long name that should not be truncated
    </h2>
    <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
      <li>
        <img alt="Placeholder image"
             fetchpriority="high"
             width="0"
             height="0"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="https://via.placeholder.com/1920x1080?w=16&amp;q=75 1x"
             src="https://via.placeholder.com/1920x1080?w=16&amp;q=75"
             style="color: transparent;"
        >
      </li>
    </ul>
  </div>
</main>
`;

exports[`Widgets/Gallery MultipleFormats smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-widgets-Gallery-Gallery-stories-module__root--O0_jV">
    <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
      Gallery
    </h2>
    <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
      <li>
        <img alt="Placeholder image"
             fetchpriority="high"
             width="0"
             height="0"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="https://via.placeholder.com/1080x1080?w=16&amp;q=75 1x"
             src="https://via.placeholder.com/1080x1080?w=16&amp;q=75"
             style="color: transparent;"
        >
      </li>
      <li>
        <img alt="Placeholder image"
             fetchpriority="high"
             width="0"
             height="0"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="https://via.placeholder.com/800x600?w=16&amp;q=75 1x"
             src="https://via.placeholder.com/800x600?w=16&amp;q=75"
             style="color: transparent;"
        >
      </li>
      <li>
        <img alt="Placeholder image"
             fetchpriority="high"
             width="0"
             height="0"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="https://via.placeholder.com/1280x720?w=16&amp;q=75 1x"
             src="https://via.placeholder.com/1280x720?w=16&amp;q=75"
             style="color: transparent;"
        >
      </li>
    </ul>
  </div>
</main>
`;

exports[`Widgets/Gallery MultipleImages smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-widgets-Gallery-Gallery-stories-module__root--O0_jV">
    <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
      Gallery
    </h2>
    <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
      <li>
        <img alt="Placeholder image"
             fetchpriority="high"
             width="0"
             height="0"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="https://via.placeholder.com/1920x1080?w=16&amp;q=75 1x"
             src="https://via.placeholder.com/1920x1080?w=16&amp;q=75"
             style="color: transparent;"
        >
      </li>
      <li>
        <img alt="Placeholder image"
             fetchpriority="high"
             width="0"
             height="0"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="https://via.placeholder.com/1920x1080?w=16&amp;q=75 1x"
             src="https://via.placeholder.com/1920x1080?w=16&amp;q=75"
             style="color: transparent;"
        >
      </li>
      <li>
        <img alt="Placeholder image"
             fetchpriority="high"
             width="0"
             height="0"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="https://via.placeholder.com/1920x1080?w=16&amp;q=75 1x"
             src="https://via.placeholder.com/1920x1080?w=16&amp;q=75"
             style="color: transparent;"
        >
      </li>
    </ul>
  </div>
</main>
`;

exports[`Widgets/Gallery Playground smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-widgets-Gallery-Gallery-stories-module__root--O0_jV">
    <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
      Gallery
    </h2>
    <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
      <li>
        <img alt="Placeholder image"
             fetchpriority="high"
             width="0"
             height="0"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="https://via.placeholder.com/1920x1080?w=16&amp;q=75 1x"
             src="https://via.placeholder.com/1920x1080?w=16&amp;q=75"
             style="color: transparent;"
        >
      </li>
    </ul>
  </div>
</main>
`;

exports[`Widgets/Gallery SingleImage smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-widgets-Gallery-Gallery-stories-module__root--O0_jV">
    <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
      Gallery
    </h2>
    <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
      <li>
        <img alt="Placeholder image"
             fetchpriority="high"
             width="0"
             height="0"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="https://via.placeholder.com/1920x1080?w=16&amp;q=75 1x"
             src="https://via.placeholder.com/1920x1080?w=16&amp;q=75"
             style="color: transparent;"
        >
      </li>
    </ul>
  </div>
</main>
`;

exports[`Widgets/Gallery SquareImage smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-widgets-Gallery-Gallery-stories-module__root--O0_jV">
    <h2 class="src-widgets-Gallery-Gallery-module__name--bagbP">
      Gallery
    </h2>
    <ul class="src-widgets-Gallery-Gallery-module__images--c5hew">
      <li>
        <img alt="Placeholder image"
             fetchpriority="high"
             width="0"
             height="0"
             decoding="async"
             data-nimg="1"
             class="src-components-Image-Image-module__image--gXjYq"
             srcset="https://via.placeholder.com/1080x1080?w=16&amp;q=75 1x"
             src="https://via.placeholder.com/1080x1080?w=16&amp;q=75"
             style="color: transparent;"
        >
      </li>
    </ul>
  </div>
</main>
`;
